lib/puppet/network/http/connection.rb

```diff
@@ -189,6 +189,32 @@ def request_with_redirects(request, options)
             request.each do |header, value|
               current_request[header] = value
             end
+          when 503, 429
+            retry_after = current_response['Retry-After']
+
+            if retry_after.nil?
+              # Set return value and advance to next loop.
+              response = current_response
+              next
+            end
+
+            begin
+              retry_sleep = Integer(retry_after)
```

Comment:
Another way of looking at it is that there would be no change from the current behavior that occurs when the agent receives a 503 with a Retry-After date.